nni/compression/pytorch/speedup/quantization_speedup/integrated_tensorrt.py

```diff
@@ -0,0 +1,134 @@
+            else:
+                layer.precision = extra_datatype
+                layer.set_output_type(0, extra_datatype)
+        # network.mark_output(model_tensors.find(ModelData.OUTPUT_NAME))
+        # Build engine and do int8 calibration.
+        engine = builder.build_cuda_engine(network)
+        return engine
+
+class TensorRt:
+    def __init__(self, model, onnx_path, input_shape, config=None, extra_layer_bit='float32', strict_datatype=False, using_calibrate=True, 
+    calibrate_type=None, calib_data=None, calibration_cache = None, batchsize=1, input_names=["actual_input_1"], output_names=["output1"]):
+        self.model = model
+        self.onnx_path = onnx_path
+        self.input_shape = input_shape
+        self.config = config
+        self.extra_layer_bit = extra_layer_bit
+        self.strict_datatype = strict_datatype
+        self.using_calibrate = using_calibrate
+        self.calibrate_type = calibrate_type
+        self.calib_data = calib_data
+        self.calibration_cache = calibration_cache
+        self.batchsize = batchsize
+        self.input_names = input_names
+        self.output_names = output_names
+        self.context = None
+
+    def tensorrt_build(self):
+        assert self.model is not None
+        assert self.onnx_path is not None
+        assert self.input_shape is not None
+
+        # convert pytorch model to onnx model and save onnx model in onnx_path
+        model_onnx = fonnx.torch_to_onnx(self.model, input_shape=self.input_shape, model_path=self.onnx_path, input_names=self.input_names, output_names=self.output_names)
+
+        if self.using_calibrate:
+            assert self.calibrate_type is not None
+            context = self.tensorrt_build_withcalib(self.onnx_path)
+        else:
+            try:
+                raise NameError('quantized without calibrate has not been supported.')
+            except NameError:
+                raise
+        self.context = context
+
+    def tensorrt_build_withcalib(self, onnx_path):
+        calib = calibrator.Calibrator(self.calib_data, self.calibration_cache, self.batchsize, self.calibrate_type)
+        engine = build_engine(onnx_path, calib, self.batchsize, self.config, self.extra_layer_bit, self.strict_datatype)
+        return engine.create_execution_context()
+
+    def inference(self, test_data):
+        elapsed_time = 0
+        inputs, outputs, bindings, stream = common.allocate_buffers(self.context.engine)
+        result = []
+        for start_idx in range(0, test_data.shape[0], self.batchsize):
+            # If the number of images in the test set is not divisible by the batch size, the last batch will be smaller.
+            # This logic is used for handling that case.
+            end_idx = min(start_idx + self.batchsize, test_data.shape[0])
+            effective_batch_size = end_idx - start_idx
+
+            # Do inference for every batch.
+            inputs[0].host = test_data[start_idx:start_idx + effective_batch_size]
```

Comment:
only change `inputs[0]`? Does `len(inputs) == 1` in this case?

Reply:
For the model with single input, `len(inputs) == 1` would always holds. This tool doesn't support model with multiple input right now.